lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -792,11 +792,15 @@ void setSuspended(ErrorInfo reason) {
          * state other than attached or attaching
          */
         synchronized void waitForSync() throws AblyException, InterruptedException {
-            boolean syncIsComplete = false; /* temporary variable to avoid potential race conditions */
-            while((channel.state == ChannelState.attached || channel.state == ChannelState.attaching) &&
-                    /* = (and not ==) is intentional */
-                    !(syncIsComplete = (!syncInProgress && syncComplete)))
+            boolean syncIsComplete = false;    /* temporary variable to avoid potential race conditions */
+            while (channel.state == ChannelState.attaching) {
                 wait();
+            }
+            if (channel.state == ChannelState.attached) {
+                while (!(syncIsComplete = (!syncInProgress && syncComplete))) {
```

Comment:
I realise you've carried this code over from work that was done before, but I'm really not comfortable with this from a readability standpoint. Could we make this code more readable? It's completely acceptable for it to span more lines if that makes it easier to understand! I don't, for example, think it would pass [InnerAssignment](https://checkstyle.sourceforge.io/config_coding.html#InnerAssignment) as/when we switch that rule on.

Reply:
I believe it's more readable than before.
I did try to make it more simplified, but ```syncIsComplete``` variable ( mentioned to avoid potential race conditions), makes it as little difficult to read. I don't know the original intention of the variable used, so I tried not to change the logic much.
Also, it's easier to understand the code, if you go through the whole waitforsync() method from the code.

Reply:
But, I will see if I can simplify it more

Reply:
I have updated with simplified implementation